server/backend/backend.go

```diff
@@ -65,9 +65,13 @@ func New(
 	clusterAddr string,
 	metrics *prometheus.Metrics,
 ) (*Backend, error) {
-	hostname, err := os.Hostname()
-	if err != nil {
-		return nil, fmt.Errorf("get hostname: %w", err)
+	hostname := conf.Hostname
+	if hostname == "" {
+		hostname, err := os.Hostname()
```

Comment:
This `hostname` value inside `if` statement shadows `hostname` in line 68.
Therefore `hostname` will not be updated.

I'll fix this in shard cluster housekeeping PR.